libnavigation-ui/src/main/java/com/mapbox/navigation/ui/speed/SpeedLimitView.java

```diff
@@ -0,0 +1,214 @@
+          MUTCD_INNER_BACKGROUND_INSET
+      );
+    } else {
+      GradientDrawable[] layers = {
+          backgroundDrawable(GradientDrawable.OVAL),
+          borderDrawable(GradientDrawable.OVAL),
+          backgroundDrawable(GradientDrawable.OVAL)
+      };
+      layerDrawable = new LayerDrawable(layers);
+      layerDrawable.setLayerInset(
+          0,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET
+      );
+      layerDrawable.setLayerInset(
+          1,
+          VIENNA_MUTCD_BORDER_INSET,
+          VIENNA_MUTCD_BORDER_INSET,
+          VIENNA_MUTCD_BORDER_INSET,
+          VIENNA_MUTCD_BORDER_INSET
+      );
+      layerDrawable.setLayerInset(
+          2,
+          VIENNA_INNER_BACKGROUND_INSET,
+          VIENNA_INNER_BACKGROUND_INSET,
+          VIENNA_INNER_BACKGROUND_INSET,
+          VIENNA_INNER_BACKGROUND_INSET
+      );
+    }
+    return layerDrawable;
+  }
+
+  @SuppressLint("DefaultLocale")
+  private String getSpeedLimit(SpeedLimitSign sign, SpeedLimitUnit unit, int speedLimitKmph) {
+    if (sign == SpeedLimitSign.MUTCD) {
+      if (unit == SpeedLimitUnit.KILOMETRES_PER_HOUR) {
+        return getContext().getString(R.string.max_speed, speedLimitKmph);
+      } else {
+        double speed = Math.round(speedLimitKmph * KILO_MILES_FACTOR);
+        String formattedSpeed = String.format("%.0f", speed);
+        return getContext().getString(R.string.max_speed, Integer.parseInt(formattedSpeed));
+      }
+    } else {
+      if (unit == SpeedLimitUnit.KILOMETRES_PER_HOUR) {
+        return String.valueOf(speedLimitKmph);
+      } else {
+        double speed = Math.round(speedLimitKmph * KILO_MILES_FACTOR);
+        return String.format("%.0f", speed);
+      }
+    }
+  }
+
+  /**
+   * Invoke to set the speed limit
+   * @param speedLimit object storing speed limit data
+   */
+  public void setSpeedLimit(@Nullable SpeedLimit speedLimit) {
+    if (speedLimit != null) {
+      Integer speedLimitKmph = speedLimit.getSpeedKmph();
+      if (speedLimitKmph != null) {
+        SpeedLimitSign speedLimitSign = speedLimit.getSpeedLimitSign();
+        speedLimitView.setImageDrawable(getSpeedDrawable(speedLimitSign));
+        SpeedLimitUnit speedLimitUnit = speedLimit.getSpeedLimitUnit();
+        String speed = getSpeedLimit(speedLimitSign, speedLimitUnit, speedLimitKmph);
+        int speedLength = speed.length();
+        SpannableStringBuilder span = new SpannableStringBuilder(speed);
+        span.setSpan(new StyleSpan(BOLD), speedLength - 2, speedLength, SPAN_EXCLUSIVE_EXCLUSIVE);
+        span.setSpan(new RelativeSizeSpan(1.8f), speedLength - 2, speedLength, 0);
+        speedLimitText.setText(span);
+      }
+    }
+  }
+
+  /**
+   * Invoke to show speed limit indicator.
+   */
+  public void show() {
+    speedView.setVisibility(VISIBLE);
```

Comment:
What happens if `setSpeedLimit` becomes `null` eventually (currently no-op) and `show` is called? It seems that an "empty" speed limit indicator will be shown. Is that expected?

Reply:
We should not maintain the state of `speedLimit`. It's up to the developer to make sure that they only invoke `show` if `speedLimit` is non null.